src/compiler/target/golang/mod.rs

```diff
@@ -0,0 +1,264 @@
+use super::*;
+
+pub struct Go {}
+
+impl Go {
+  /// Generate golang code using tokens from parser
+  pub fn generate(lb: &mut LangBuilder, t: AnilizedTokens) -> Result<(), LocationError> {
+    // TODO: Replace when file importing is implemented
+    lb.code("package main");
+
+    let mut code = Self {};
+
+    // define functions
+    for (_, func) in t.functions {
+      code.function(func, lb);
+    }
+
+    // define types
+    for (_, type_) in t.types {
+      code.code_type(type_, lb);
+    }
+
+    // define structs
+    for (_, structure) in t.structs {
+      code.structure(structure, lb);
+    }
+
+    // define globals
+    for (_, glob) in t.vars {
+      code.global_var(glob, lb);
+    }
+
+    Ok(())
+  }
+  /// Get the type as a string
+  pub fn get_type(&mut self, type_: Option<Type>) -> String {
```

Comment:
Maybe we can better use the language builder here instead of returning a string so we can better format the code.

Reply:
Surely this would build it on a new line rather than appending to the current one. If this is true then it would fail to compile into valid Go because Go breaks statements at new lines.

Reply:
I think it's probably best to just append to the current line using the Inline builder, using a new line here would output wired code i think.  
With inline structs it probably best to add a Block to the Inline builder so we can still have multi line structs in functions like this:
```go
func foo(bar: struct {
  Foo string
  Bar string
}) {}
```

Reply:
I've tried a few times now, but changing this is breaking a lot of functionality. Most importantly this makes it harder to create a prefix. Maybe instead of returning a string, we could return an `Inline` rather than appending to lang builder in `get_type`? I'm not 100% confident with the lang builder so I may be missing something, but I just can't get it to work and still generate the prefixes properly 😞

Reply:
NVM I've made some good progress, but do have one question. 

Is there any reason why the `langbuilder`'s `function` method should refuse to append some code? I'm asking because I was using this for the inline structs, but even though there are no errors, the code is not showing up in the results.

Reply:
That could be because of this: https://github.com/talpalang/talpa/blob/master/src/compiler/target/builder.rs#L147  
Currently we ignore Code blocks within inlines but we have to also support that to make this work

Reply:
```rust
Self::Inline(items) => {
        let mut out: Vec<String> = vec![];
        for item in items {
          out.push(item.get_line(builder));
        }
        out.join("")
      }
      Self::Block(_, _) => String::new(), // TODO this is not yet used inside a inline but i would like to also support this
```

Just spotted this. I think this will need to be implemented for us to work with this. I'll continue working on the golang implementation, but it won't work 100% until this is implemented.

Reply:
sure, that seems like a good idea.

Reply:
Cool, almost done. Just need to run some tests